### ISSUE TITLE
[BUG] Javascript getters calling during config initialization

### ISSUE THREAD

**Issue by bublig737** [node-config/node-config#749]
**Describe the bug**
// config/default.js

```
module.exports = {
    application: {
        name: process.env.NAME,
        get currentIp() {
            console.log('Getter called')
            return getCurrentIp()
        }
    }
}

function getCurrentIp() {
    return '123.123.123'
}
```

//index.js
```
console.log('Log BEFORE config import');

const config = require('config')

console.log('Log AFTER config import');

console.log('Current Ip', config.application.currentIp);
```
Logs output:
Log BEFORE config import
Getter called
Getter called
Getter called
Getter called
Getter called
Log AFTER config import
Getter called
Current Ip 123.123.123

**Expected behavior**

Expected to not calling getters during initialization of config, as in commonJs. For me its needed because I'm using
dynamic cashed values so getters need to be called only after some services will be inited.

**Screenshots**
<img width="943" alt="image" src="https://github.com/node-config/node-config/assets/31284223/fa01a5bc-4548-4266-9c67-a5cec2fd2656">

<img width="946" alt="image" src="https://github.com/node-config/node-config/assets/31284223/9193b64e-ab92-499c-a4a3-a9ce32ad8906">

### Please tell us about your environment:

 -  node-config version: 3.3.9
 -  node-version: v16.17.0

**Comment by jdmarshall** [comment]
How does it behave with `defer()`?

https://github.com/node-config/node-config/wiki/Special-features-for-JavaScript-configuration-files#deferred-values-in-javascript-configuration-files

**Comment by jdmarshall** [comment]
@markstos Are you sure this is a bug? Building the data in CONFIG in config.js involves traversing the data returned by each individual config file, merging it, and then marking it as read-only. I think it's actually expected behavior that the fields in a .js or .mjs file would be accessed multiple times during the load process.

But what I think is wanted here is a defer() or async load call. I think this is working as designed.
